### ISSUE TITLE
The "directory" field in the compile_commands.json is not respected when executing the compiler executable

### ISSUE THREAD

**Issue by mdzahidh** [microsoft/vscode-cpptools#3413]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Linux, Fedora 28
- VS Code Version: 1.32.3
- C/C++ Extension Version: 0.22.1
- A clear and concise description of what the bug is.

Consider a compile_commands.json item:
```
[
    {
        "directory": "some_intended_directory",
        "arguments": [
            "/some_path/some_wrapper_script_for_compiler.sh", .....
        ]
    }
]
```
We noticed that when when the compiler script is executed, it chdirs into "/some_path/" but not the "directory" field, which should have been the correct behavior.
This causes failures/crashes when cpp-tools parses through our compile_commands.json.

**To Reproduce**

Cook up a compile_commands.json like the one above, and it will repro all the time. 

**Expected behavior**
Considering the above compile_commands.json example:
chdir("some_intended_directory")
exec("/some_path/some_wrapper_script_for_compiler.sh")

**Comment by bobbrow** [comment]
Sorry for the delay in responding.  We missed this issue in our triage passes. 😞 

Just to clarify, the issue here is that the cwd is not set to what your script expects it to be?  The extension is actually running your script, correct?  I'm looking at the code and I do see us prepending the 'directory' to arg[0] when it is presumed to be a relative path.

**Comment by mdzahidh** [comment]
when we strace'ed, we found that the current directory is considered to be the directory part of the full path to the script, but not the "directory" key. In our case, the script assumes the current directory to be set to exactly the "directory" key, and therefore fails. We just thought, its a bug on cpptool's side because it doesn't seem to be respecting the "directory" key where its present.

**Comment by Colengms** [comment]
I believe this issue was addressed in v1.0.0 of the extension.  Closing as fixed.
